app/src/main/java/com/example/my/player/presentation/ui/PlayerActivity.kt

```diff
@@ -0,0 +1,108 @@
+package com.example.my.player.presentation.ui
+
+import android.content.Context
+import android.os.Build
+import android.os.Bundle
+import androidx.annotation.RequiresApi
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.ViewModelProvider
+import com.bumptech.glide.Glide
+import com.bumptech.glide.load.resource.bitmap.RoundedCorners
+import com.example.my.R
+import com.example.my.databinding.ActivityAudioPlayerBinding
+import com.example.my.player.presentation.state.TrackViewState
+import com.example.my.player.presentation.utils.DisplayPx
+import com.example.my.player.presentation.view_model.PlayerViewModel
+import com.example.my.search.presentation.ui.SearchActivity
+
+class PlayerActivity : AppCompatActivity() {
+
+    private lateinit var binding: ActivityAudioPlayerBinding
+    private lateinit var viewModel: PlayerViewModel
+
+   private var track: TrackViewState? = null
+
+    @RequiresApi(Build.VERSION_CODES.TIRAMISU)
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityAudioPlayerBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        viewModel = ViewModelProvider(
+            this,
+            PlayerViewModel.getFactory()
+        ).get(PlayerViewModel::class.java)
+
+        setupTrackInfoObserver()
+
+        track = intent.getParcelableExtra(
+            SearchActivity.Companion.TRACK_DATA,
+            TrackViewState::class.java
+        )
+
+        if (track != null) {
+            viewModel.loadTrackInfo(track)
+            displayTrackInfo(track!!)
+        }
+
+         viewModel.playerStateLiveData.observe(this) {
+             changeButtonIcon(it == PlayerViewModel.Companion.STATE_PLAYING)
+             enableButton(it != PlayerViewModel.Companion.STATE_DEFAULT)
+         }
+
+         viewModel. currentTrack.observe(this) {
+             binding.playingTime.text = it as CharSequence?
```

Comment:
Если нужно безопасно вызвать `as`, то правильным вариантом будет:
```suggestion
             binding.playingTime.text = it as? CharSequence
```